applications/ConvectionDiffusionApplication/python_scripts/coupled_structural_thermal_solver.py

```diff
@@ -141,19 +143,30 @@ def AdvanceInTime(self, current_time):
         return new_time
 
     def InitializeSolutionStep(self):
-        self.structural_solver.InitializeSolutionStep()
         self.thermal_solver.InitializeSolutionStep()
+        self.thermal_solver.Predict()
+
+        KratosMultiphysics.Logger.PrintInfo("\n" + "\t" + "Solving Thermal part..." + "\n")
+        self.thermal_solver.SolveSolutionStep()
```

Comment:
this step should also be done in `SolveSolutionStep`

Reply:
I am doing it in this way since the predict of the mechanical part should be with the updated temperature, otherwise we cannot "converge" since the residual was badly computed

Reply:
Hm I see

My issue with this is that it does not fit the intended design of the AnalysisStage

@RiccardoRossi I think you can evaluate best how this should be done

Reply:
In this particular case what I'd do is to implement a do nothing `InitializeSolutionStep` and `Predict` and do everything, first for the thermal case and then for the structural one within the `SolveSolutionStep`.

Reply:
Done @rubenzorrilla

Reply:
Please, comment the code explaining why this is done in such a way.